airflow/models.py

```diff
@@ -910,16 +910,34 @@ def is_queueable(
         elif self.state == State.QUEUED and not include_queued:
             return False
         # is the task runnable and have its dependencies been met?
-        elif (
-                self.state in State.runnable() and
-                self.are_dependencies_met(
-                    ignore_depends_on_past=ignore_depends_on_past,
-                    flag_upstream_failed=flag_upstream_failed)):
+        elif (self.state in State.runnable() and
+                  self.are_dependencies_met(
+                      ignore_depends_on_past=ignore_depends_on_past,
+                      flag_upstream_failed=flag_upstream_failed) and
+                  not self.is_superceded_by_future()):
             return True
         # anything else
         else:
             return False
 
+    @provide_session
+    def is_superceded_by_future(self, session):
+        if not self.task.only_run_latest:
+            return False
+        logging.info("checking for future")
+        TI = TaskInstance
+        ti = session.query(TI).filter(
+            TI.dag_id == self.dag_id,
+            TI.task_id == self.task_id,
+            TI.state == State.SUCCESS,
+            TI.execution_date > self.execution_date,
+        ).first()
+        if ti:
+            if self.state in State.runnable():
+                self.set_state(State.FUTURE_SUCCEEDED, session)
```

Comment:
Having an is_ method change state is a little unintuitive. @bolkedebruin just did some scheduler work to clean up this kind of pattern. Can you split the is_ method from the set call, so that change state is explicit, rather than done via what appears to be a read method?